src/components/MapSearchBar/index.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react'
+import Autosuggest from 'react-autosuggest'
+import PropTypes from 'prop-types'
+import { Icon } from 'semantic-ui-react'
+import './MapSearchBar.css'
+
+class MapSearchBar extends React.Component {
+  static propTypes = {
+    setLocation: PropTypes.func.isRequired,
+    recenterMap: PropTypes.func.isRequired,
+    config: PropTypes.object.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      value: '',
+      placeholder: 'Search for an address or a place',
+      suggestions: []
+    }
+
+    this.onChangeAutosuggest = this.onChangeAutosuggest.bind(this)
+    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this)
+    this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this)
+    this.getSuggestionValue = this.getSuggestionValue.bind(this)
+    this.renderSuggestion = this.renderSuggestion.bind(this)
+    this.makeRequest = this.makeRequest.bind(this)
+    this.clearSearch = this.clearSearch.bind(this)
+    this.renderClearButton = this.renderClearButton.bind(this)
+  }
+
+  // Will be called every time you need to recalculate suggestions
+  onSuggestionsFetchRequested ({value}) {
+    if (value.length >= 2) {
+      this.autocomplete(value)
+    }
+  }
+
+  // Will be called every time you need to set suggestions to []
+  onSuggestionsClearRequested () {
+    this.setState({
+      suggestions: []
+    })
+  }
+
+  // Teach Autosuggest what should be input value when suggestion is clicked
+  getSuggestionValue (suggestion) {
+    const lat = suggestion.geometry.coordinates[1]
+    const lng = suggestion.geometry.coordinates[0]
+    const latlng = [lat, lng]
+    const name = suggestion.properties.label
+
+    // Stores latlng and name of selected location in Redux
+    this.props.setLocation(latlng, name)
+    // Recenters map to the selected location's latlng
+    this.props.recenterMap(latlng)
+    return suggestion.properties.label
+  }
```

Comment:
This is a `get` function and should just return a label. Setting Redux state here is a side-effect, so let's move that part out to the [`onSuggestionSelected` handler](https://github.com/moroshko/react-autosuggest#onSuggestionSelectedProp).